### DISCUSSION TITLE
Need descriptor for in memory proto schema

### DISCUSSION THREAD

**Discussion by vickydahiya** [square/wire#2318] [Q&A]
I have an in-memory proto schema which I am reading from the schema registry. I am looking for a way or the best way if there are more than one, to get the descriptor for this schema. How can I do that? I am new to protobuf and pardon me if the provided information is not sufficient, but I can provide any required additional details.

**Comment by oldergod** [comment]
You could use [SchemaEncoder](https://github.com/square/wire/blob/master/wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/internal/SchemaEncoder.kt) and do something like
```kotlin
    val schemaEncoder = SchemaEncoder(schema)

    val fileDescriptorSet =  FileDescriptorSet.newBuilder()
      .addAllFile(schema.protoFiles.map {
        FileDescriptorProto.parseFrom(
          schemaEncoder.encode(it).toByteArray()
        )
      })
      .build()
```

I'm not sure why `SchemaEncoder` is under an `internal` package but it should work.
